lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -39,11 +39,17 @@ LinearEncodingAttr toLinearEncoding(Attribute layout, ArrayRef<int64_t> shape) {
 }
 
 unsigned getTotalElemsPerThread(Attribute layout, ArrayRef<int64_t> shape) {
+  if (auto distLayout = mlir::dyn_cast<DistributedEncodingTrait>(layout)) {
+    return distLayout.getTotalElemsPerThread(shape);
```

Comment:
so it returns a different value that what the linear layout does? This means you have a distributed layout that is not a linear layout?